File: iosApp/iosApp/GiftContents.swift
import SwiftUI
import shared

final class GiftContents {
    static func attention(_ width: CGFloat, _ height: CGFloat, _ image: String, _ isTablet: Bool = false, _ navigateTo: () -> some View) -> some View {
        let screenSize = UISize(width: width, height: height)
        let frameSize = UIConfig.Home.companion.getAttentionFrameSize(deviceSize: screenSize, isTablet: isTablet)
        let bottomBarSize = UIConfig.Home.companion.getBottomBarFrame(frameSize: frameSize)
        return NavigationLink(destination: {
            navigateTo()
        }, label: {
            UIDraw.rcFrame(frameSize.toCGSize(), shadow: 3, shadowX: 3, shadowY: 3) {
                ZStack {
                    Image(image)
                        .resizable()
                        .clipShape(RoundedRectangle(cornerRadius: 15))
                    Rectangle()
                        .foregroundStyle(.white.opacity(0.25))
                        .frame(width: bottomBarSize.width, height: bottomBarSize.height)
                        .frame(maxHeight: .infinity, alignment: .bottom)
                        .clipShape(RoundedRectangle(cornerRadius: 15), style: FillStyle())
                        .blur(radius: 3.0)
                }
            }
        })
    }
    static func largeAttention(_ width: CGFloat, _ height: CGFloat, _ image: String, _ navigateTo: () -> some View) -> some View {
        let screenSize = UISize(width: width, height: height)
        let frameSize = UIConfig.Home.companion.getLargeAttentionFrameSize(deviceSize: screenSize)
        let bottomBarSize = UIConfig.Home.companion.getBottomBarFrame(frameSize: frameSize)
        return NavigationLink(destination: {
            navigateTo()
        }, label: {
            UIDraw.rcFrame(frameSize.toCGSize(), shadow: 3, shadowX: 3, shadowY: 3) {
                ZStack {
                    Image(image)
                        .resizable()
                        .clipShape(RoundedRectangle(cornerRadius: 15))
                    Rectangle()
                        .foregroundStyle(.white.opacity(0.25))
                        .frame(width: bottomBarSize.width, height: bottomBarSize.height)
                        .frame(maxHeight: .infinity, alignment: .bottom)
                        .clipShape(RoundedRectangle(cornerRadius: 15), style: FillStyle())
                        .blur(radius: 3.0)
                }
            }
        })
    }
    static func history(_ width: CGFloat, _ height: CGFloat, _ image: String, _ isTablet: Bool = false, _ navigateTo: () -> some View) -> some View {
        let screenSize = UISize(width: width, height: height)
        let frameSize = UIConfig.History.companion.getContentFrameSize(deviceSize: screenSize, isTablet: isTablet)
        let imageSize = UIConfig.History.companion.getContentIconFrameSize(deviceSize: frameSize, isTablet: isTablet)
        return NavigationLink(destination: {
            navigateTo()
        }, label: {
            UIDraw.rcFrame(frameSize.toCGSize(), color: .white, shadow: 3, shadowX: 3, shadowY: 3) {
                HStack(spacing: 5) {
                    Image(image)
                        .resizable()
                        .scaledToFit()
                        .clipShape(RoundedRectangle(cornerRadius: 15))
                    VStack(alignment: .leading, spacing: 5) {
                        UIDraw.text("ダミー", color: .black, style: "Bold")
                        UIDraw.text("ダミーテキスト", color: .black)
                        Spacer()
                    }
                    
                }
                .frame(maxWidth: .infinity, alignment: .leading)
                .padding(10)
            }
            .overlay {
                RoundedRectangle(cornerRadius: 15)
                    .stroke(lineWidth: 1.0)
                    .foregroundStyle(.black)
            }
        })
    }
}

File: androidApp/src/main/java/com/cistus/hunter/android/GiftContents.kt
package com.cistus.hunter.android

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.runtime.Composable
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.saveable.rememberSaveable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.blur
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.unit.dp
import com.cistus.hunter.UIConfig
import com.cistus.hunter.UISize
import com.cistus.hunter.toDpSize

class GiftContents {
    companion object {
        @Composable
        fun DrawAttention(width: Double, height: Double, image: Int, isTablet: Boolean = false,
                          navigateTo: @Composable (Float) -> Unit) {
            val screenSize = UISize(width, height)
            val frameSize = UIConfig.Home.getAttentionFrameSize(screenSize, isTablet)
            val bottomBarSize = UIConfig.Home.getBottomBarFrame(frameSize)
            val showDialog = rememberSaveable { mutableStateOf(false) }

            UIDraw.DrawRCFrame(frameSize.toDpSize(),
                modifier = Modifier.shadow(3.dp, shape = RoundedCornerShape(15.dp)),
                onTapped = { showDialog.value = true }) {
                Box(modifier = Modifier.fillMaxSize()) {
                    Image(painterResource(image), "")
                    UIDraw.CustomColumn(style = "Bottom", modifier = Modifier.blur(5.dp)) {
                        Box(modifier = Modifier
                            .width(bottomBarSize.width.dp)
                            .height(bottomBarSize.height.dp)
                            .background(Color(1f, 1f, 1f, .25f)))
                    }
                }
            }
            if (showDialog.value)
                UIDraw.DrawDialog(showDialog, "Back", true) {
                    navigateTo(it)
                }
        }
        @Composable
        fun DrawLargeAttention(width: Double, height: Double, image: Int,
                               navigateTo: @Composable (Float) -> Unit) {
            val screenSize = UISize(width, height)
            val frameSize = UIConfig.Home.getLargeAttentionFrameSize(screenSize)
            val bottomBarSize = UIConfig.Home.getBottomBarFrame(frameSize)
            val showDialog = rememberSaveable { mutableStateOf(false) }

            UIDraw.DrawRCFrame(frameSize.toDpSize(), modifier = Modifier.shadow(3.dp),
                onTapped = { showDialog.value = true }) {
                Box(modifier = Modifier.fillMaxSize()) {
                    Image(painterResource(image), "")
                    UIDraw.CustomColumn(style = "Bottom", modifier = Modifier.blur(5.dp)) {
                        Box(modifier = Modifier
                            .width(bottomBarSize.width.dp)
                            .height(bottomBarSize.height.dp)
                            .background(Color(1f, 1f, 1f, .25f)))
                    }
                }
            }
            if (showDialog.value)
                UIDraw.DrawDialog(showDialog, "Back", true) {
                    navigateTo(it)
                }
        }
        @Composable
        fun DrawHistory(width: Double, height: Double, image: Int, isTablet: Boolean = false,
                        navigateTo: @Composable (Float) -> Unit) {
            val screenSize = UISize(width, height)
            val frameSize = UIConfig.History.getContentFrameSize(screenSize, isTablet)
            val imageSize = UIConfig.History.getContentIconFrameSize(screenSize, isTablet)
            val showDialog = rememberSaveable { mutableStateOf(false) }
            Box {
                UIDraw.DrawRCFrame(frameSize.toDpSize(), color = Color.White,
                    modifier = Modifier.border(1.dp, color = Color.Black, shape = RoundedCornerShape(15.dp))
                        .shadow(3.dp, shape = RoundedCornerShape(15.dp)),
                    onTapped = { showDialog.value = true }) {
                    UIDraw.CustomRow(style = "CenterStart", spacing = 5.dp,
                        modifier = Modifier.padding(5.dp)) {
                        Image(painterResource(image), "",
                            modifier = Modifier.width(imageSize.width.dp).height(imageSize.height.dp)
                                .padding(5.dp).clip(RoundedCornerShape(15.dp)))
                        UIDraw.CustomColumn(style = "TopStart") {
                            UIDraw.DrawText("ダミー", color = Color.Black, style = "Bold")
                            UIDraw.DrawText("ダミーテキスト", color = Color.Black)
                        }
                    }
                }
            }
            if (showDialog.value)
                UIDraw.DrawDialog(showDialog, closeText = "Back", true) {
                    navigateTo(it)
                }
        }
    }
}